widget/button.go

```diff
@@ -82,7 +82,7 @@ func (b *buttonRenderer) Refresh() {
 			b.icon = canvas.NewImageFromResource(b.button.Icon)
 			b.objects = append(b.objects, b.icon)
 		} else {
-			b.icon.Resource = b.button.Icon
+			b.icon = canvas.NewImageFromResource(b.button.Icon)
```

Comment:
Unfortunately this does not work - the button never refreshes.
The previous code was setting the data on the icon, this is creating a new one.
If creating a new one is the right thing to do then the canvasobject must be added to the object draw list (and the old one removed) - somewhat like 3 lines earlier